### PR TITLE
pythonPackages.intelhex init at 2.1

### PR DESCRIPTION
**Pull request by pjones**
###### Motivation for this change

This package is a dependency of another package that I'll be submitting shortly.  In other words, I need this PR approved so I can submit another one :)

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
1. see the Nixpkgs manual on contributing Python packages.
2. the manual and the header of this file state where expressions should go
3. fetchPypi instead of fetchurl
4. when disabling tests, always leave a comment explaining why.
5. meta is missing

**Comment by pjones** [comment]
@FRidh Sorry about that, I got super lazy on this one.  It's (hopefully) fixed now.

File: pkgs/development/python-modules/intelhex/default.nix [hunk 1]
```diff
@@ -13,13 +13,6 @@ buildPythonPackage rec {
     sha256 = "0k5l1mn3gv1vb0jd24ygxksx8xqr57y1ivgyj37jsrwpzrp167kw";
   };
 
-  # FAIL: test_xrange_longint (intelhex.test.TestXrangeLongInt)
-  # AssertionError: OverflowError not raised
-  #
-  # Harmless test failure:
-  #  https://github.com/bialix/intelhex/issues/14
-  doCheck = false;
-
```
Comment (Mic92) [inline pkgs/development/python-modules/intelhex/default.nix]: Lets see if travis confirms it.

Comment (pjones) [inline pkgs/development/python-modules/intelhex/default.nix]: It's my understanding that the test will only fail on 64-bit systems.  Please keep that in mind.

**Comment by FRidh** [comment]
pushed https://github.com/NixOS/nixpkgs/commit/aa3f37ceed582c8cb6e871db528b7216d17b474f

**Comment by pjones** [comment]
@FRidh The sha256 you used for the patch is bad:

~~~
output path ‘/nix/store/b0rbpdjhacn5p96h91hwan2pm9hf67h7-f251aef214daa2116e15ff7f7dcec1639eb12d5b.patch’ has sha256 hash ‘1fiqbha923idibjzdyq9p6xrjwak36yr02pzb7dg8lf1mv0w1cjc’ when ‘02i15qjmcz7mwbwvyj3agl5y7098rag2iwypdilkaadhbslsl9b9’ was expected
~~~

**Comment by FRidh** [comment]
ugh that's the last time I use `fetchpatch`. 
Fixed in f2b97e061a

**Comment by pjones** [comment]
Thanks for the quick fix!

**Pull request closed without merging**